api/ccapi/include/optimizer.h

```diff
@@ -50,7 +50,7 @@ class Optimizer {
    * @brief     get Optimizer Type
    * @retval    Optimizer type
    */
-  virtual OptimizerType getType() = 0;
```

Comment:
I guess now both are valid??

Reply:
Integer value will be dynamic for custom layer so I think we could only retrieve the integer from the context.

Reply:
Yes, given enums exist, there must be a method to get that enum value given the optimizer object.
The return type can be integer or OptimizerType, it's about readability as it does not matter with enum.